publish-release.sh

```diff
@@ -46,26 +57,108 @@ case "$choice" in
 template_version=${version}.${status}
 
+# Config checks for stable releases.
+
+if [ "${status}" == "stable" -a "${skip_stable}" == "0" ]; then
+  echo "Publishing a stable release. Checking that configuration is valid to perform stable release specific steps."
+
+  read -p "Enter personal access token (GH_TOKEN) for godotengine/godot: " personal_gh_token
+  if [[ "${personal_gh_token}" != "github_pat_"* ]]; then
+    echo "Provided personal access token should start with 'github_pat', aborting."
+    exit 1
+  fi
+
+  if ! gh api repos/godotengine/godot/git/refs/tags | grep -q ${godot_version}; then
+    echo "The tag '${godot_version}' does not exist in godotengine/godot, aborting."
+    echo "Push commits and create it manually before running this script."
+    exit 1
+  fi
+
+  if [ ! -d "${UPLOAD_STEAM_PATH}" ]; then
+    echo "Invalid config.sh: UPLOAD_STEAM_PATH is not a directory, aborting."
+    exit 1
+  fi
+
+  if [ "${latest_stable}" == "1" ]; then
+    if [ ! -d "${UPLOAD_EGS_PATH}" ]; then
+      echo "Invalid config.sh: UPLOAD_EGS_PATH is not a directory, aborting."
+      exit 1
+    fi
+    if [ ! -x "${UPLOAD_ITCH_BUTLER}" ]; then
+      echo "Invalid config.sh: UPLOAD_ITCH_BUTLER does not point to an executable, aborting."
+      exit 1
+    fi
+  fi
+fi
+
 # Upload to GitHub godot-builds
 
+echo "Uploading release to to godotengine/godot-builds repository."
+
 if [ -z "${GODOT_BUILDS_PATH}" ]; then
   echo "Missing path to godotengine/godot-builds clone in config.sh, necessary to upload releases. Aborting."
   exit 1
 fi
 
 ${GODOT_BUILDS_PATH}/tools/upload-github.sh -v ${version} -f ${status}
 
-# Web editor
+# Stable release only
 
-scp -P 22 -r web/${template_version} ${WEB_EDITOR_HOSTNAME}:/home/akien/web_editor/
-sleep 2
-command="sudo mv /home/akien/web_editor/${template_version} /var/www/editor.godotengine.org/public/releases/"
-command="${command}; cd /var/www/editor.godotengine.org; sudo chown -R www-data:www-data public/releases/${template_version}"
-command="${command}; sudo ./create-symlinks.sh -v ${template_version}"
-if [ $web_editor_latest == 1 ]; then
-  command="${command} -l"
+if [ "${status}" == "stable" -a "${skip_stable}" == "0" ]; then
+  namever=Godot_v${godot_version}
+
+  echo "Uploading stable release to main GitHub repository."
+
+  export GH_TOKEN=${personal_gh_token}
+  pushd git
+  # Get release details from existing godot-builds release.
+  release_info=$(gh release view ${godot_version} --repo godotengine/godot-builds --json name,body -q '.name, .body')
+  read -r RELEASE_TITLE RELEASE_DESC <<< "$release_info"
+
+  gh release create ${godot_version} --repo godotengine/godot --title "$RELEASE_TITLE" --notes "$RELEASE_DESC"
+  gh release upload ${godot_version} ${reldir}/[Gg]* ${reldir}/mono/[Gg]*
+  # Concatenate SHA sums.
+  cp ${reldir}/SHA512-SUMS.txt .
+  cat ${reldir}/mono/SHA512-SUMS.txt >> SHA512-SUMS.txt
+  gh release upload ${godot_version} SHA512-SUMS.txt
+  rm SHA512-SUMS.txt
+  popd
```

Comment:
This copies the release info from godotengine/godot-builds and uploads it anew to godotengine/godot.

As we don't give @GodotBuilder write permissions on godotengine/godot for security, this requires one of us to get a PAT for upload. I suggest making one with 1 day expiry each time this is needed, which is why the script asks for it at the start (so it's only pasted in the terminal and not hardcoded anywhere).

Reply:
I need to recheck this, as it didn't seem to work fully in https://github.com/godotengine/godot/releases/tag/4.4.1-stable - I've now copied the text from the godot-builds release manually, but initially the description was empty.

Also, we should handle the `${latest_stable}` flag here too to make sure that only the latest stable gets flagged as latest release on GitHub.

Reply:
> I need to recheck this, as it didn't seem to work fully in https://github.com/godotengine/godot/releases/tag/4.4.1-stable - I've now copied the text from the godot-builds release manually, but initially the description was empty.

Found a fix using `jq`.

> Also, we should handle the `${latest_stable}` flag here too to make sure that only the latest stable gets flagged as latest release on GitHub.

Seems like we might be able to rely on GH's default logic here:
```
$ gh release create --help | grep Latest
      --latest                       Mark this release as "Latest" (default: automatic based on date and version)
```
Since the default is to guess, and there's no way to say "not latest", I guess we don't really have much options :D